### PR TITLE
Fixes skvh table with audit only option does not work properly 

### LINKED ISSUE #56: skvh table with audit only option does not work properly
When creating a skvh table and if the history is not given  in the option we get an error as it tries to create the name of the history table in the trigger function using list_to_existing_atom. This fails.

### PR DESCRIPTION
**Pull request by shamis**
Fixes #56

### CODE REVIEW

**Comment by c-bik** [comment]
Can you add eunit tests for different table create options with read/write checks?

**Comment by shamis** [comment]
@c-bik Have added the test cases for read and write for skvh tables without both audit and history and without history.

File: src/imem_datatype.erl [hunk 1]
```diff
@@ -15,7 +15,7 @@
                                'bxor','bsl','bsr','abs','div','rem',
                                'min','max','float','now','date','element',
                                'size','bit_size','byte_size','binary_part',
-                               'phash2','md5','throw']).
+                               'phash2','md5','throw','hd']).
 -define(UNSAFE_ERLANG_FUNCTIONS,['list_to_atom','binary_to_atom','list_to_pid','binary_to_term','is_pid','is_port','is_process_alive']).
 
 -define(ROWFUN_EXTENSIONS,[{<<"nodef">>,1}
```
Comment (c-bik) [inline src/imem_datatype.erl]: This is a generic change so please add `tl` too, take out of PR/branch and push to master right away with CM "added safe operators 'hd' and 'tl'"

File: src/imem_dal_skvh.erl [hunk 2]
```diff
@@ -461,9 +461,13 @@ drop_table(Channel) when is_binary(Channel) ->
 
 build_aux_table_info(Table) ->
 	["","",Channel,"","","",""] = imem_meta:parse_table_name(Table),
-	HistoryTable = ?HIST_FROM_STR(Channel),
+    HistoryTable = try ?HIST_FROM_STR(Channel) of
+        H -> [H]
+    catch
+        _:_ -> []
```
Comment (c-bik) [inline src/imem_dal_skvh.erl]: Instead of `[]` return `['$no_history']`

File: src/imem_datatype.erl (hunk 1 above)
Comment (shamis) [inline src/imem_datatype.erl]: @c-bik have done this

File: src/imem_datatype.erl [hunk 3]
```diff
@@ -16,6 +16,7 @@
                                'min','max','float','now','date','element',
                                'size','bit_size','byte_size','binary_part',
                                'phash2','md5','throw','hd','tl']).
+
```
Comment (c-bik) [inline src/imem_datatype.erl]: Can you remove this one empty line change too?

File: src/imem_dal_skvh.erl [hunk 4]
```diff
@@ -461,9 +461,13 @@ drop_table(Channel) when is_binary(Channel) ->
...
+    end,
 	{AuditTable,TransTime} = audit_table_time(Channel),
-    {AuditTable,HistoryTable,TransTime,Channel}.
+    {[AuditTable|HistoryTable],TransTime,Channel}.
```
Comment (c-bik) [inline src/imem_dal_skvh.erl]: Can you try:
```erlang
build_aux_table_info(Table) ->
	["","",Channel,"","","",""] = imem_meta:parse_table_name(Table),
    HistoryTable = try ?HIST_FROM_STR(Channel) of H -> H
                   catch _:_ -> '$no_history'
                   end,
	{AuditTable,TransTime} = audit_table_time(Channel),
    {[AuditTable,HistoryTable],TransTime,Channel}.
```

File: include/imem_dal_skvh.hrl [hunk 5]
```diff
@@ -15,17 +15,17 @@
...
            "    end"),
-           ?ADDIF([audit, history], __Opts,
+           ?ADDIF([history], __Opts,
            ",\n    case lists:member(no_history,TrOpts) of \n"
            "        true ->  ok;\n"
-           "        false -> ok = imem_dal_skvh:write_history(HistoryTable,AuditInfoList)\n"
+           "        false -> ok = imem_dal_skvh:write_history(hd(HistoryTable),AuditInfoList)\n"
            "    end"),
            (fun(_E) ->
                     if length(_E) == 0 -> "";
```
Comment (c-bik) [inline include/imem_dal_skvh.hrl]: Can you try:
```erlang
-define(ADDIF(__Flags,__Opts,__Code),
        (fun(_F,_O) ->
                 case lists:usort([_F,_O]) of
                     _O -> __Code;
                     _ -> ""
                end
        end)(__Flags,lists:usort(__Opts))).
-define(skvhTableTrigger(__Opts, __ExtraFun),
        list_to_binary(
          ["fun(OldRec,NewRec,Table,User,TrOpts) ->\n"
           "    start_trigger",
           ?ADDIF([audit], __Opts,
           ",\n    {[AuditTable,HistoryTable],TransTime,Channel}\n"
           "        = imem_dal_skvh:build_aux_table_info(Table),\n"
           "    AuditInfoList = imem_dal_skvh:audit_info(User,Channel,AuditTable,TransTime,OldRec,NewRec),\n"
           "    case lists:member(no_audit,TrOpts) of \n"
           "        true ->  ok;\n"
           "        false -> ok = imem_dal_skvh:write_audit(AuditInfoList)\n"
           "    end"),
           ?ADDIF([audit,history], __Opts,
           ",\n    case lists:member(no_history,TrOpts) of \n"
           "        true ->  ok;\n"
           "        false -> ok = imem_dal_skvh:write_history(HistoryTable,AuditInfoList)\n"
           "    end"),
           (fun(_E) ->
                    if length(_E) == 0 -> "";
                       true -> ",\n    "++_E end
            end)(__ExtraFun),
           "\n    %extend_code_start"
           "\n    %extend_code_end"
           "\nend."])
       ).
```

Comment (shamis) [inline include/imem_dal_skvh.hrl]: @c-bik have made few changes

**Pull request merged**